### ISSUE TITLE
ESC7 - Error when attempting to add an officer

### ISSUE THREAD

**Issue by jsdhasfedssad** [ly4k/Certipy#33]
When attemting to exploit ESC7 and the account I use for authentication does not have the right Manage Certificates, I must add that account as a new officer in order to grant the account that right. This fails for me using Certipy 2.0.6.
![esc7_dc1](https://user-images.githubusercontent.com/97032878/155497732-3422646e-68c6-4840-b8d1-82875a84f290.png)

**Comment by ly4k** [comment]
Yes, you can remove the CA officer afterwards with `-remove-officer`. The request fails because you're trying to connect to the DC. You should connect to the CA

**Comment by jsdhasfedssad** [comment]
I actually tried connecting to the ADCS server when I tested this but that failed as well :)
![esc7_server2](https://user-images.githubusercontent.com/97032878/155555452-c0cf8ee0-5f67-42ad-a22c-49481fe521a7.png)

**Comment by ly4k** [comment]
The second image makes more sense to me. It's a mistake in my implementation. I accidentally assumed the CA and the DC was on the same server, which results in Certipy trying to connect to LDAP on the CA, or connecting to the CA DCOM on the DC. Thanks for reporting this. I'll look into a fix soon

**Comment by ly4k** [comment]
Should be fixed in 2.0.8. Can you please verify?

**Comment by jsdhasfedssad** [comment]
Using 2.0.8 I can now add the officer. Good! 
![esc7_3](https://user-images.githubusercontent.com/97032878/155710730-2a9e2d77-663c-4502-828d-bad58b1051b5.png)

The next step in my case is to enable the certificate template SubCA but that fails for me.
![esc7_2](https://user-images.githubusercontent.com/97032878/155710752-be9fd8df-1f1f-4f66-a45b-f77335c898cd.png)

**Comment by ly4k** [comment]
Should be fixed in 2.0.9. The new LDAP DNS resolution was not applied for the `-enable-template`. Can you verify that it is working now? I really appreciate that you report these issues. Thanks!

**Comment by jsdhasfedssad** [comment]
Great! This works now. I actually managed to execute the complete ESC7 attack this time. Don't worry about asking me to verify fixes, I will have so much fun owning my client's infrastructure using your tool :D No more hassle having to own a domain-joined machine, bypass AV, AMSI, Applocker, execution policies, language constrained mode and what not then having to upload and import powershell scripts! All I need now is essentially Responder, Hashcat, Bloodhound and Certipy :D

**Comment by ly4k** [comment]
Great. Haha yes, Active Directory itself is full of attack vectors
